pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy

```diff
@@ -216,38 +153,57 @@ public class ModelInterpreter implements Serializable {
                     throw e
                 }
             }
-        }
+        }.call()
     }
 
+    /**
+     * Execute a body closure within a "withEnv" block.
+     *
+     * @param envVars A list of "FOO=BAR" environment variables. Can be null.
+     * @param body The closure to execute
+     * @return The return of the resulting executed closure
+     */
     def withEnvBlock(List<String> envVars, Closure body) {
         if (envVars != null && !envVars.isEmpty()) {
             return {
                 script.withEnv(envVars) {
                     body.call()
                 }
-            }
+            }.call()
```

Comment:
But that means that you return the return value of the `call` not the closure itself. So that would indicate to me that something is funky somewhere else.